modules/context/context.go

```diff
@@ -210,7 +211,7 @@ func Contexter() macaron.Handler {
 			if err == nil && len(repo.DefaultBranch) > 0 {
 				branchName = repo.DefaultBranch
 			}
-			prefix := setting.AppURL + path.Join(url.QueryEscape(ownerName), url.QueryEscape(repoName), "src", "branch", branchName)
+			prefix := setting.AppURL + path.Join(util.PathEscapeSegments(ownerName), util.PathEscapeSegments(repoName), "src", "branch", branchName)
```

Comment:
`ownerName` and `repoName` should be `url.PathEscape` and `branchName` should be escaped with `util.PathEscapeSegments`

Reply:
done